scripts/building/check_enforcer.ps1

```diff
@@ -129,29 +121,78 @@ function sdk_cli_checks([ref]$failed_reason_ref, $valid_status_array) {
     }
 }
 
-function run_checks() {
-
-    $need_to_check = New-Object System.Collections.Generic.HashSet[string]
-    git diff --name-only HEAD origin/main | ForEach-Object {
-        if ($snippet_debug -eq 1) {
-            Write-Output "git diff --name-only HEAD main $_"
+function trigger_prepare($input_paths, [ref]$failed_reason_ref) {
+    foreach ($input_path in $input_paths) {
+        if ($checks.ContainsKey("samples_connections_connection")) {
+            continue
+        }
+        if ($input_path.Contains("examples") -or $input_path.Contains("samples")) {
+            # Define the input path
+            Push-Location $GithubWorkspace
+            $pipelines_samples = (python $GithubWorkspace/scripts/readme/readme.py -c | ConvertFrom-Json)
```

Comment:
if one PR changes template, for example, remove path filters, then check enforcer will not guard any more?

Reply:
Of course~ this is expected.